### PR TITLE
Fix API Installer to generate unique Operation IDs

### PR DESCRIPTION
**Pull request by mbohlool**
OperationIDs generated by api_installer.go may not be unique and that causes long auto-generated function names in swagger generated clients. This is a step toward making those IDs unique.

Reference Issue kubernetes/features#53 and #13414

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/33337)

### CODE REVIEW

**Comment by mbohlool** [comment]
Note: The PR labeled as "size/L" because of the generated code. It is not actually that big (first commit is the manual one).

File: api/swagger-spec/v1.json [hunk 1]
```diff
@@ -15402,7 +15402,7 @@
       "type": "string",
       "method": "GET",
       "summary": "connect GET requests to proxy of Service",
-      "nickname": "connectGetNamespacedServiceProxy",
+      "nickname": "connectGetNamespacedServiceProxyWithPath",
```
Comment (lavalamp) [inline api/swagger-spec/v1.json]: It's unfortunate that we have to split out every request to the proxy separately by verb. :/

File: api/swagger-spec/v1.json [hunk 2]
```diff
@@ -13193,7 +13193,7 @@
       "type": "versioned.Event",
       "method": "GET",
       "summary": "watch individual changes to a list of Secret",
-      "nickname": "watchNamespacedSecretList",
+      "nickname": "watchNamespacedSecretListForAllNamespaces",
```
Comment (lavalamp) [inline api/swagger-spec/v1.json]: I'm not sure how critical the first "Namespaced" is. E.g., maybe (watchNamespacedSecretList, watchNamespacedSecretListForAllNamespaces) could instead be (watchSecretList, watchSecretListForAllNamespaces)?

(It's in there to distinguish between cluster and namespace scoped resources. I'm not sure if we have any with colliding names any more. @deads2k?)

Comment (lavalamp) [inline api/swagger-spec/v1.json]: If it is important to keep the Namespaced at the beginning, maybe the second version could be watchAllNamespacesSecretList? That would keep the length down a little bit, although it's a bit less natural english.

**Review by lavalamp — COMMENTED** [review]
The other thing to think about is if we should include group and version--that would future-proof these names a bit.

**Comment by lavalamp** [comment]
I think my comments are optional :)

**Comment by mbohlool** [comment]
About adding group and version, I think clientset would prevent that. We are generating one client per group/version. Do you suggest we have one client for all and add group/version to the name?

File: api/swagger-spec/v1.json (hunk 2 above)
Comment (mbohlool) [inline api/swagger-spec/v1.json]: Removing first "Namespaced" make more sense to me. At some point I thought about it, but I am not sure why I didn't go with it.

File: api/swagger-spec/v1.json (hunk 1 above)
Comment (mbohlool) [inline api/swagger-spec/v1.json]: it is outside of the scope of this change, I will let it pass for this PR but will think about it.

File: pkg/apiserver/api_installer.go [hunk 3]
```diff
@@ -47,10 +47,11 @@ type APIInstaller struct {
...
 type action struct {
-	Verb   string               // Verb identifying the action ("GET", "POST", "WATCH", PROXY", etc).
-	Path   string               // The path of the action
-	Params []*restful.Parameter // List of parameters associated with the action.
-	Namer  ScopeNamer
+	Verb          string               // Verb identifying the action ("GET", "POST", "WATCH", PROXY", etc).
+	Path          string               // The path of the action
+	Params        []*restful.Parameter // List of parameters associated with the action.
+	Namer         ScopeNamer
+	allNamespaces bool // true iff the action is namespaced but works on aggregate result for all namespaces
```
Comment (smarterclayton) [inline pkg/apiserver/api_installer.go]: Should be public, like other fields on this type.

**Comment by smarterclayton** [comment]
One comment from me

**Comment by lavalamp** [comment]
Ah, good point re: group/version. I forgot we were doing each client separately.

File: pkg/apiserver/api_installer.go (hunk 3 above)
Comment (mbohlool) [inline pkg/apiserver/api_installer.go]: Done.

**Comment by mbohlool** [comment]
I've addressed all comments. I am still thinking about group/version (for another PR). Please LGTM this one if it looks good.

For group/version, I was thinking about supporting a customization method in spec generation code and use it only for root swagger spec to add group/version to the name of operation. I will send another PR for that.

**Comment by lavalamp** [comment]
LGTM. Customization for the universal doc sounds good, too.

**Comment by mbohlool** [comment]
Only change was gofmt. applying LGTM again.

**Pull request merged**